### PR TITLE
Login

### PR DESCRIPTION
**Pull request by Vivekrajput20**

### CODE REVIEW

File: css/signup.css [hunk 1]
```diff
@@ -234,6 +238,44 @@ body {
 	font-weight: 700;
 	margin-bottom: 5px;
 }
+.login-message{
+	color: #155724;
+	background-color: #d4edda;
+	border-radius: 2px;
+padding:15px 0;
```
Comment (ankitkataria) [inline css/signup.css]: indent

File: css/signup.css [hunk 2]
```diff
@@ -234,6 +238,44 @@ body {
...
+
+}
+.login-message a {
+	color: #155724;
+	font-weight: 600;
+	text-decoration: none;
+
```
Comment (ankitkataria) [inline css/signup.css]: remove the new lines

File: webpages/login.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    </head>
+    <body>
+         <?php
+            $email = $password = "";
+            $emailerr = $passerr = "";
+            function prepare($data){
+            
+              $data = trim($data);
+              $data = stripslashes($data);
+               $data = htmlspecialchars($data);
```
Comment (ankitkataria) [inline webpages/login.php]: indent, be consistent as to where you put the closing or the opening braces, with the functions as well as the if statements.

File: webpages/login.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                            $_SESSION["email"] = $user["email"];
+                            $_SESSION["fname"] = $user["fname"];
+                            $_SESSION["lname"] = $user["lname"];
+                            header('Location: /webpages/home.php');
+                        }
+                        else{
+                            $passerr = "<div class=\"error\"> Wrong Email or Password ! </div>";
+
+                        }
+
```
Comment (ankitkataria) [inline webpages/login.php]: remove unused lines

File: webpages/db.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?php
+$servername = $username = $passd = "";
```
Comment (ankitkataria) [inline webpages/db.php]: add this file to .gitignore file, and put dummy values in this file which the users can change to test on their systems. read more about gitignore from https://git-scm.com/docs/gitignore

File: webpages/login.php (hunk 4 above)
Comment (Vivekrajput20) [reply webpages/login.php]: done

**Comment by fristonio** [comment]
Set up a more intuitive directory structure like this
```
.
|-- config
|-- public
|   |-- includes
|   |   `-- signup.php
|   |-- index.php
|   `-- static
|       |-- css
|       |-- images
|       `-- scripts
`-- schema
```

**Comment by fristonio** [comment]
change index.html to index.php

File: css/signup.css [hunk 6]
```diff
@@ -15,19 +15,21 @@
   unicode-range: U+0460-052F, U+20B4, U+2DE0-2DFF, U+A640-A69F;
 }
 /*basic reset*/
-* {margin: 0; padding: 0;}
+* {margin: 0; padding: 0;
```
Comment (fristonio) [inline css/signup.css]: format properly

File: css/signup.css [hunk 7]
```diff
@@ -116,8 +112,11 @@ body {
 	padding: 20px 30px;
 	box-sizing: border-box;
 	width: 80%;
+<<<<<<< Updated upstream
 	margin: 0 10%;
-	
+=======
```
Comment (fristonio) [inline css/signup.css]: remove these lines, you have to manually remove such conflicts in git

File: webpages/login.php [hunk 8]
```diff
@@ -0,0 +1,92 @@
+<?php
+session_start();
+if (!empty($_SESSION["uid"]))
+{
```
Comment (fristonio) [inline webpages/login.php]: use format for if as
```php
if (condition) {
  block;
}
```
if the block have a single statement do not use curly braces at all, it increases readability

File: webpages/sample.config.php [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<?php
+$servername = $username = $passd = "";
```
Comment (fristonio) [inline webpages/sample.config.php]: remove this

File: webpages/signup.php [hunk 10]
```diff
@@ -1,157 +1,211 @@
...
+            }
+            
+                }
+                if(empty($_POST["pass"]))
+                {
+                  $passerr = "<div class=\"error\">  * Password is required! </div>";
+                }
+              else{
+                $password = prepare($_POST["pass"]);
+                if (strlen((string)$password)<6)
```
Comment (fristonio) [inline webpages/signup.php]: do not typecast $password to string rather check first if it is a string and then check if the length is <6

**Review by fristonio — APPROVED** [review]

**Pull request merged**